devito/finite_differences/operations.py

```diff
@@ -0,0 +1,173 @@
+import sympy
+import numpy as np
+
+import devito
+from devito.symbolics.search import retrieve_functions
+from devito.symbolics.extended_sympy import FrozenExpr
+
+__all__ = ['Mul', 'Add', 'Pow']
+
+
+def fd_parameters(obj):
+    """
+    Process a set of Mul, Add, Pow expression and outputs
+    the space_order, time_order of the whole expression.
```

Comment:
again, let's think about this properly: do we actually need to distinguish between ``space`` and ``time``? and why?

Reply:
Probably not actually just another dimension so should be able to generate fd same way as space